src/core/storage/compression/Benchmark.cpp

```diff
@@ -0,0 +1,71 @@
+#include <core/storage/compression/Compressor.h>
+#include <core/storage/compression/Decompressor.h>
+namespace fs = std::filesystem;
+
+bool exists(const fs::path &p, fs::file_status s = fs::file_status{}) {
+	return fs::status_known(s) ? fs::exists(s) : fs::exists(p);
+}
+
+void check_initial_compressed_size(const std::string &initial_file_folder, const std::string &compressed_file) {
+	unsigned long initial_size;
+	std::cout << "directory: " << fs::is_directory(initial_file_folder) << std::endl;
+	if (fs::is_directory(initial_file_folder))
+		initial_size = std::accumulate(
+			fs::recursive_directory_iterator(initial_file_folder.c_str()),
+			fs::recursive_directory_iterator(), 0,
+			[](auto sz, auto entry) { return is_directory(entry) ? sz : sz + file_size(entry); });
+	else
+		initial_size = fs::file_size(initial_file_folder);
+
+	unsigned long long int compressed_size = fs::file_size(compressed_file);
+
+	std::cout << std::endl << std::endl
+	          << "#############################################################" << std::endl;
+	std::cout << "Passed with initial size: " << initial_size << " and compressed size: "
+	          << compressed_size << std::endl;
+	std::cout << "#############################################################" << std::endl
+	          << std::endl << std::endl;
+}
+
+bool clean(const std::map<std::string, std::string> &files) {
+	return std::ranges::all_of(files.cbegin(), files.cend(),
+	                           [](const auto &pair) {
+		                           return fs::remove_all(pair.second);
+	                           });
+}
+
+int main() {
```

Comment:
Although this implementation looks fine, I think using something like [google-benchmark](https://github.com/google/benchmark) would be better.

Moreover, after adding conan this is as simple as adding `benchmark/1.6.0` in `conanfile.txt`.

Reply:
It will be in the next pull request because I need Conan which is in the main branch already